app/components/filter-plaques/component.js

```diff
@@ -1,21 +1,53 @@
 import Ember from 'ember';
 
-export default Ember.Component.extend({
+var ID_FILTERS = ['contributors', 'funders', 'publishers'];
 
+export default Ember.Component.extend({
+    filters: null,
     init () {
-        this._super(...arguments);
-        let parameters = this.get('parameters');
-        this.set('filters', Object.keys(parameters).map((key, idx, arr) => {
-            return {
-                "key": key,
-                "value": parameters[key]
-            }
-        }).filter((item) => {
-            if (item.key == "page") {
-                return false;
-            }
-            return true;
-        }));
+      this._super(...arguments);
+      let parameters = this.get('parameters');
+      var filters = Object.keys(parameters).filter((key) => {
+        return key !== "page";
+      }).map((key) => {
+        return {
+          "key": key,
+          "value": parameters[key]
+        }
+      });
+
+      // Find and format all query params that use ids for the value
+      var ids = filters.filter((item) => {
+          return ID_FILTERS.indexOf(item.key) !== -1;
+      }).map((param) => {
+          return {key: param.value}
+      });
+
+      // Fetch display names
+      if (ids.length > 0) {
+        this.fetchAgentDetails(ids).then((data) => {
+            var displayFilters = filters.map((filter) => {
+              var value = filter.value;
+              data.forEach((agentData) => {
+                if (value === agentData.id) {
+                  value = agentData.name;
```

Comment:
That wouldn't fix it, because the second time around is where it breaks, and then theres the issue of if the id occurs twice. 

I think what we need here is a function that gets passed an id, and returns a name, that we can map over.
The other part is that we need to be certain that we're only modifying plaques that have ids on them, Then we have the guarantee that we only set each once, and that it is an id.